R/write_output.R

```diff
@@ -0,0 +1,329 @@
+      )
+      write_parquet(samples, samples_path)
+      cli::cli_alert_success("Wrote samples to {.path {samples_path}}")
+
+      # Process and write summarized quantiles
+      summaries_path <- file.path(
+        output_dir,
+        job_id,
+        "summaries",
+        paste0(task_id, ".parquet")
+      )
+      write_parquet(summaries, summaries_path)
+      cli::cli_alert_success("Wrote summaries to {.path {summaries_path}}")
+
+
+      # Write EpiNow2 model
+      model_path <- file.path(
+        output_dir,
+        job_id,
+        "tasks",
+        task_id,
+        "model.rds"
+      )
+      saveRDS(fit, model_path)
+      cli::cli_alert_success("Wrote model to {.path {model_path}}")
+
+
+      # Write model run metadata
+      metadata_path <- file.path(
+        output_dir,
+        job_id,
+        "tasks",
+        task_id,
+        "metadata.json"
+      )
+      jsonlite::write_json(metadata, metadata_path, pretty = TRUE)
+      cli::cli_alert_success("Wrote metadata to {.path {metadata_path}}")
+    },
+    error = function(cnd) {
+      # Downgrade erroring out to a warning so we can catch and log
+      cli::cli_warn(
+        "Failure writing outputs",
+        parent = cnd,
+        class = "no_outputs"
+      )
+    }
+  )
+
+  invisible(NULL)
+}
+
+#' Create output directory structure for a given job and task.
+#'
+#' This function generates the necessary directory structure for storing output
+#' files related to a job and its tasks, including directories for raw samples
+#' and summarized quantiles.
+#'
+#' @param output_dir String. The base output directory path.
+#' @param job_id String. The identifier for the job.
+#' @param task_id String. The identifier for the task.
+#'
+#' @return The path to the base output directory (invisible).
+#' @export
+write_output_dir_structure <- function(output_dir, job_id, task_id) {
+  # Top level
+  dir.create(output_dir,
+    showWarnings = FALSE
+  )
+  # Second level
+  dir.create(file.path(output_dir, job_id),
+    showWarnings = FALSE
+  )
+  # Third level
+  dir.create(file.path(output_dir, job_id, "tasks"),
+    showWarnings = FALSE
+  )
+  dir.create(file.path(output_dir, job_id, "samples"),
+    showWarnings = FALSE
+  )
+  dir.create(file.path(output_dir, job_id, "summaries"),
+    showWarnings = FALSE
+  )
+
+  # Fourth level
+  dir.create(file.path(output_dir, job_id, "tasks", task_id),
+    showWarnings = FALSE
+  )
+
+  invisible(output_dir)
+}
+
+#' Extract posterior draws from a Stan fit object.
+#'
+#' This function extracts posterior draws for specific parameters from a Stan
+#' fit object and prepares a fact table containing unique date-time-parameter
+#' combinations for further merging.
+#'
+#' @param fit A Stan fit object with posterior estimates.
+#'
+#' @return A list containing two elements: `stan_draws` (the extracted draws in
+#' long format) and `fact_table` (a table of unique date-time-parameter
+#' combinations).
```

Comment:
I don't think we should. The fact table is an internal implementation detail, not anything exposed to the user. We only need to do it as a workaround because getting dates from EpiNow2 is hard.